### PR TITLE
[v7r0] Add option to retry update of hosts

### PR DESCRIPTION
**Pull request by petricm**
BEGINRELEASENOTES
*FrameworkSystem
NEW: Add option to retry to update hosts in dirac-admin-update-instance

ENDRELEASENOTES

### CODE REVIEW

File: FrameworkSystem/scripts/dirac-admin-update-instance.py [hunk 1]
```diff
@@ -219,11 +225,20 @@ def updateInstance(version, hosts):
   for host in restartFail:
     gLogger.notice(" - %s" % host)
   gLogger.notice("XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX")
+
+  if retry > 0:
+    retryHosts = list(set(updateFail + restartFail))
+    gLogger.notice("Retrying update on (%s atempts remaining):" % retry)
```
Comment (fstagni) [inline FrameworkSystem/scripts/dirac-admin-update-instance.py:231]:
```suggestion
    gLogger.notice("Retrying update on (%s attempts remaining):" % retry)
```

File: FrameworkSystem/scripts/dirac-admin-update-instance.py [hunk 2]
```diff
@@ -179,12 +184,13 @@ def restartHost(hostName):
...
 
 
-def updateInstance(version, hosts):
+def updateInstance(version, hosts, retry):
   """
   Update each server of an instance and restart them
 
   :param version: version vArBpC you want to update to
   :param hosts: list of hosts to be updated
+  :param retry: number of retry attempts on hosts that have failed to update
```
Comment (fstagni) [inline FrameworkSystem/scripts/dirac-admin-update-instance.py:193]:
```suggestion
  :param int retry: number of retry attempts on hosts that have failed to update
```
(also add the type also for the other parameters, also in other functions)

File: FrameworkSystem/scripts/dirac-admin-update-instance.py [hunk 3]
```diff
@@ -23,22 +23,28 @@
                        )
 Script.registerSwitch("", "hosts=", "Comma separated list of hosts or file containing row wise list of hosts"
                                     " targeted for update (leave empty for all)")
+Script.registerSwitch("", "retry=", "Number of retry attempts on hosts that have failed to update")
```
Comment (fstagni) [inline FrameworkSystem/scripts/dirac-admin-update-instance.py:26]: Don't you want to add default of 1 retry?

Comment (petricm) [reply FrameworkSystem/scripts/dirac-admin-update-instance.py:26]: As the retry is an optional flag, I would opt for having it set to 0. Along the lines "explicit over implicit".

**Review by fstagni — APPROVED** [review]

**Review by atsareg — APPROVED** [review]

**Pull request merged**